src/Library.qs

```diff
@@ -19,63 +19,59 @@
         DataSize : Int);
 
     //Internal QRam type for a single QRAM Bit (is composed to form a full memory block)
-    internal newtype SingleBitQRAM = (Lookup : ((Qubit[], Qubit) => Unit is Adj + Ctl), 
-        AddressSize : Int);
+    internal newtype SingleBitQRAM = (Lookup : ((Qubit[], Qubit[]) => Unit is Adj + Ctl));
+
+    function ImplicitQRAMOracle(DataValues : (Bool[],Bool[])[]) : QRAM {
+        mutable addressSize = 0;
+        mutable valueSize = 0;
+        for((_,(address,value)) in Enumerated(DataValues)){
```

Comment:
Since you are dropping/not using the index generated with `Enumerated` then you can just do
```suggestion
        for ((address, value)  in DataValues){
```